nat-lab/tests/test_reconnections.py

```diff
@@ -109,9 +109,9 @@ async def test_mesh_reconnect(
             await ping(beta_connection, alpha.ip_addresses[0], 15)
 
         await client_alpha.simple_start()
-        await client_alpha.set_meshmap(api.get_meshmap(alpha.id))
 
         await asyncio.gather(
+            client_alpha.set_meshmap(api.get_meshmap(alpha.id)),
```

Comment:
This doesn't really fix the race condition. But on the other hand, starting to wait for event before action, might fix the race condition:

```
        async with run_async_context(
            asyncio.gather(
                client_alpha.wait_for_event_peer(
                    beta.public_key, [telio.State.Connected]
                ),
                client_alpha.wait_for_event_on_any_derp([telio.State.Connected]),
            )
        ) as events:
            await client_alpha.set_meshmap(api.get_meshmap(alpha.id))
            await events
```

Reply:
oh nice, thanks for taking a look :pray: 

Fixed.